src/Countries/US/Michigan/MichiganCityTaxes/MichiganCityTax.php

```diff
@@ -0,0 +1,243 @@
+            } elseif ($this->resident_city_no_work_no_tax_works_in_2_other_cities_that_have_tax) {
+                $this->percentage_worked = $this->tax_information->primary_nonresident_city_percentage_worked;
+                return self::DEFAULT_TAX_RATE;
+            }
+        } elseif ($this->tax_information->secondary_nonresident_city === $this->getCityName()) {
+            if ($this->resident_city_is_primary_work_city_with_tax_works_in_different_city_with_tax && $this->isSpecialCity($this->tax_information->secondary_nonresident_city)) {
+                return $this->getSpecialCityRate($this->tax_information->resident_city, $this->tax_information->secondary_nonresident_city, false);
+            } elseif ($this->resident_city_is_primary_work_city_with_tax_works_in_different_city_with_tax) {
+                return self::DEFAULT_TAX_RATE;
+            } elseif ($this->resident_city_no_work_has_tax_works_in_2_other_cities_that_have_tax && $this->isSpecialCity($this->tax_information->secondary_nonresident_city)) {
+                if ($this->tax_information->secondary_nonresident_city_percentage_worked > $this->tax_information->primary_nonresident_city_percentage_worked) {
+                    $this->percentage_worked = $this->tax_information->secondary_nonresident_city_percentage_worked;
+                    return $this->getSpecialCityRate($this->tax_information->resident_city, $this->tax_information->secondary_nonresident_city, false);
+                }
+            } elseif ($this->resident_city_no_work_has_tax_works_in_2_other_cities_that_have_tax) {
+                if ($this->tax_information->secondary_nonresident_city_percentage_worked > $this->tax_information->primary_nonresident_city_percentage_worked) {
+                    $this->percentage_worked = $this->tax_information->secondary_nonresident_city_percentage_worked;
+                    return self::DEFAULT_TAX_RATE;
+                }
+            } elseif ($this->resident_city_no_work_no_tax_works_in_2_other_cities_that_have_tax  && $this->isSpecialCity($this->tax_information->secondary_nonresident_city)) {
+                return $this->getSpecialCityRate($this->tax_information->primary_nonresident_city, $this->tax_information->secondary_nonresident_city, false);
+            } elseif ($this->resident_city_no_work_no_tax_works_in_2_other_cities_that_have_tax) {
+                $this->percentage_worked = $this->tax_information->secondary_nonresident_city_percentage_worked;
+                return self::DEFAULT_TAX_RATE;
+            }
+        }
+    }
+
+    public function getSpecialCityRate($resident_city, $work_city, $is_resident_city)
+    {
+        $resident_city_resident_rate = 0;
+        $resident_city_nonresident_rate = 0;
+        $work_city_nonresident_rate = 0;
+
+        if ($resident_city === DetroitTax::getCityName()) {
```

Comment:
if all the taxes extend `MichiganCityTax` could you not just call `$this->getResidencyTaxRate()` etc and then not need all the checks or am I reading this wrong?

Reply:
So the tax can come through 3 different ways, as a resident city tax, primary work city tax, or secondary city tax. So that is why there is so much checking